lxd/storage/drivers/driver_pure_util.go

```diff
@@ -1001,6 +1103,22 @@ func (d *pure) getMappedDevPath(vol Volume, mapVolume bool) (string, revert.Hook
 	case pureModeISCSI:
 		diskPrefix = "scsi-"
 		diskSuffix = fmt.Sprintf("%s::%s", vol.pool, volName)
+	case pureModeNVMe:
+		diskPrefix = "nvme-eui."
+
+		pureVol, err := d.client().getVolume(vol.pool, volName)
+		if err != nil {
+			return "", nil, err
+		}
+
+		if len(pureVol.Serial) <= 10 {
```

Comment:
what is the magic number 10 here?

Reply:
Yes, comment is missing.

For iscsi, when volumes are mounted, they will have entire volume name in the device suffix "<pod>::<volume_name>".

However, for nvme/tcp this is not the case. Instead, device name contains last 10 characters from the volume serial.